scala/scala.bzl

```diff
@@ -293,52 +293,47 @@ def write_manifest(ctx):
         output=ctx.outputs.manifest,
         content=manifest)
 
-def _write_launcher(ctx, rjars, main_class, jvm_flags, args, run_before_binary, run_after_binary):
-    classpath = ':'.join(
-      ["$JAVA_RUNFILES/{repo}/{spath}".format(
-          repo = ctx.workspace_name, spath = f.short_path
-      ) for f in rjars])
-
-    location_expanded_jvm_flags = []
-    for jvm_flag in jvm_flags:
-        location_expanded_jvm_flags.append(ctx.expand_location(jvm_flag, ctx.attr.data))
-
-    content = """#!/bin/bash
-
-case "$0" in
-/*) self="$0" ;;
-*)  self="$PWD/$0" ;;
-esac
-
-if [[ -z "$JAVA_RUNFILES" ]]; then
-  if [[ -e "${{self}}.runfiles" ]]; then
-    export JAVA_RUNFILES="${{self}}.runfiles"
-  fi
-  if [[ -n "$JAVA_RUNFILES" ]]; then
-    export TEST_SRCDIR=${{TEST_SRCDIR:-$JAVA_RUNFILES}}
-  fi
-fi
-
-export CLASSPATH={classpath}
-{run_before_binary}
-$JAVA_RUNFILES/{repo}/{java} {jvm_flags} {main_class} {args} "$@"
-BINARY_EXIT_CODE=$?
-{run_after_binary}
-exit $BINARY_EXIT_CODE
-""".format(
-        classpath = classpath,
-        repo = ctx.workspace_name,
-        java = ctx.executable._java.short_path,
-        jvm_flags = " ".join(location_expanded_jvm_flags),
-        main_class = main_class,
-        args = args,
-        run_before_binary = run_before_binary,
-        run_after_binary = run_after_binary,
-    )
+def _write_launcher(ctx, rjars, main_class, jvm_flags, args="", preamble=""):
+    classpath = ":".join([
+        "${JAVA_RUNFILES}/%s/%s" % (ctx.workspace_name, j.short_path)
+        for j in rjars
+    ])
+    jvm_flags = " ".join([
+        ctx.expand_location(f, ctx.attr.data) for f in jvm_flags
+    ])
+    javabin = "/".join([
+        "${JAVA_RUNFILES}", ctx.workspace_name, ctx.executable._java.short_path
+    ])
+    template = ctx.attr._java_stub_template.files.to_list()[0]
+
+    command = " ".join([
+        "sed",
+        "\n".join([
+            # Insert the preamble at the first blank line
```

Comment:
Can you add a comment that it has to be after #! /bin/bash ? (and that would implicitly explain why less exotic prepend solutions don't quite work)

It is a subtlety I missed upon first review.